worker/uniter/runner/runner.go

```diff
@@ -421,9 +432,16 @@ func (runner *runner) runCharmHookWithLocation(hookName, charmLocation string, r
 			v, _ := env[k]
 			return v
 		}
+		osEnv = func() []string {
```

Comment:
It doesn't feel great that we reimplement all of this for getEnv and now osEnv in 2 places.
And our HookVars function is growing enough attributes that its probably time to start refactoring it into a class instead of just a function.